### PR TITLE
[release/v2.15] Leader election config in ComponentSettings (#6641)

### PR DESCRIPTION
**Pull request by irozzo-1A**
This is an automated cherry-pick of #6641

/assign irozzo-1A

```release-note
Provide the possibility of configuring leader election parameters for user cluster components.
```

### CODE REVIEW

File: pkg/validation/cluster.go [hunk 1]
```diff
@@ -464,3 +464,22 @@ func ValidateUpdateWindow(updateWindow *kubermaticv1.UpdateWindow) error {
 	}
 	return nil
 }
+
+func ValidateLeaderElectionSettings(l kubermaticv1.LeaderElectionSettings) error {
```
Comment (xmudrii) [inline pkg/validation/cluster.go]: I don't see that this function is used anywhere? Is that as expected?

Comment (irozzo-1A) [reply pkg/validation/cluster.go]: Yep, we had no cluster validation webhook in, 2.15. But you are right we can remove this.

File: pkg/validation/cluster_test.go [hunk 2]
```diff
@@ -23,6 +23,7 @@ import (
 	"testing"
 
 	kubermaticv1 "k8c.io/kubermatic/v2/pkg/crd/kubermatic/v1"
+	"k8s.io/utils/pointer"
```
Comment (xmudrii) [inline pkg/validation/cluster_test.go]: This seems to be leftover.

**Review by xmudrii — APPROVED** [review]
/lgtm

**Pull request merged**